lib/ClangImporter/ClangImporter.cpp

```diff
@@ -6565,6 +6565,17 @@ static bool hasDestroyTypeOperations(const clang::CXXRecordDecl *decl) {
   return false;
 }
 
+static bool hasCustomCopyOrMoveConstructor(const clang::CXXRecordDecl *decl) {
+  // std::pair and std::tuple might have copy and move constructors, but that
+  // doesn't mean they are safe to use from Swift, e.g. std::pair<UnsafeType, T>
+  if (decl->isInStdNamespace() &&
+      (decl->getName() == "pair" || decl->getName() == "tuple")) {
```

Comment:
Could we be more specific here?

Reply:
I don't think we should check fully qualified names if that's what you mean. For instance in libc++ pair is actually `std::__1::pair`, other stdlibs would have different fully qualified names.
Do you ideas on how to make this condition more reliable?